src/utils/pgwire/src/pg_message.rs

```diff
@@ -628,12 +629,17 @@ impl<'a> BeMessage<'a> {
             }
 
             BeMessage::ErrorResponse(error) => {
+                use thiserror_ext::AsReport;
                 // For all the errors set Severity to Error and error code to
                 // 'internal error'.
 
                 // 'E' signalizes ErrorResponse messages
                 buf.put_u8(b'E');
-                let msg = error.to_string();
+                let msg = if env_var_is_true("RW_PRETTY_ERROR") {
+                    format!("{:#}", error.as_ref().as_report())
+                } else {
+                    format!("{}", error.as_ref().as_report())
+                };
```

Comment:
Why would we ever want the `Compact` behavior? I prefer not to offer too many choices unless there's a good reason.

Reply:
Besides, can we add a test (maybe expect test) for an error message?

Reply:
> Why would we ever want the `Compact` behavior?

This could be a temporary workaround. 😕 SQL logic tests and planner tests are heavily relying on the error message, and I don't want to mess them into this PR.

Also, I'm not sure if the developers writing SQL logic tests are intended to match the full messages (instead of the interesting part), like below. If so, then `Pretty` will be less friendly.

https://github.com/risingwavelabs/risingwave/blob/761130cad1756c5b8cffa8e8379e66aa9b311fe2/e2e_test/source/basic/ddl.slt#L21
https://github.com/risingwavelabs/risingwave/blob/761130cad1756c5b8cffa8e8379e66aa9b311fe2/e2e_test/ddl/invalid_operation.slt#L267

Reply:
> Besides, can we add a test (maybe expect test) for an error message to demonstrate the error format?

There's one in `thiserror-ext`. Do you want another one for RisingWave's real cases? Then I'll write an integration test connecting to the `pgwire` server.

Reply:
> This could be a temporary workaround. 😕 SQL logic tests and planner tests are heavily relying on the error message, and I don't want to mess them into this PR.

If DX is the only concern or trouble, I think we should try to fix it now or later. At least pretty should be the default behavior. 

I think it's ok to update tests in this PR, as the core changes are very small. 

If you don't want to do it now, maybe we should introduce `RW_COMPACT_ERROR`, instead of `RW_PRETTY_ERROR`.

> Do you want another one for RisingWave's real cases? 

Yes, that's preferred. So that we can prevent either `risingwave` or `thiserror-ext`'s unexpected changes.

Reply:
Update error messages in `slt` and pretty is always enabled now.

Reply:
> > Do you want another one for RisingWave's real cases?
> 
> Yes, that's preferred. So that we can prevent either `risingwave` or `thiserror-ext`'s unexpected changes.

I find the error message is not good enough... Maybe let's do this in future PRs. 🤡

- Manually implemented `caused by`:
https://github.com/risingwavelabs/risingwave/blob/e0d45005cb75efce942fe677cf46b6d42292083f/src/frontend/src/binder/expr/mod.rs#L71-L76
- RPC boundary: #13282

Reply:
It's OK if the error message is not good. Just have a `expect-test` to let us know what's the current status is enough. 🤣 And we can see how it's improved in future PRs.

Reply:
I find writing tests with madsim is the easiest way, but the udf example does not work there. 🤡

Reply:
🤡